### PR TITLE
Implementing connection pooling for HTTP clients for Mediaflux::Http::Request

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Advances #180

### CODE REVIEW

File: app/models/mediaflux/http/request.rb [hunk 1]
```diff
@@ -60,27 +60,40 @@ def self.default_xml_namespace_uri
...
-        @http_client = http_client || Net::HTTP::Persistent.new
-        # https is not working correctly on td-meta1 we should not need this, but we do...
-        @http_client.verify_mode = OpenSSL::SSL::VERIFY_NONE
-
+        @http_client = http_client || self.class.find_or_create_http_client
         @file = file
         @session_token = session_token
+
+        finalizer = self.class.create_finalizer
+        ObjectSpace.define_finalizer(self, finalizer)
```
Comment (jrgriffiniii) [inline app/models/mediaflux/http/request.rb:90]: This implements the following pattern:

```ruby
class Foo
  def initialize(data_needed_for_finalization)
    ObjectSpace.define_finalizer(self, self.class.create_finalizer(data_needed_for_finalization))
  end

  def self.create_finalizer(data_needed_for_finalization)
    proc {
      puts "finalizing #{data_needed_for_finalization}"
    }
  end
end
```

https://ruby-doc.org/3.3.0/ObjectSpace.html#method-c-define_finalizer

Comment (jrgriffiniii) [reply app/models/mediaflux/http/request.rb:90]: I now see two errors here, I am very sorry.

**Comment by jrgriffiniii** [comment]
https://github.com/pulibrary/tiger-data-app/pull/524 has this passing for me locally (I was struggling to understand which changes I introduced which resulted in the error found for https://app.circleci.com/pipelines/github/pulibrary/tiger-data-app/1458/workflows/2f912149-8f46-408d-b15b-d3a7b456697b/jobs/6833). Hopefully this is resolved on CircleCI, and I can then open this PR for review.

File: app/models/mediaflux/http/request.rb [hunk 2]
```diff
@@ -2,6 +2,22 @@
...
+        def self.create_finalizer
+          proc {
+            @http_client.shutdown unless @http_client.nil?
+          }
+        end
+
+        def initialize(**args)
+          super
+
+          finalizer = self.class.create_finalizer
```
Comment (carolyncole) [inline app/models/mediaflux/http/request.rb:16]: @jrgriffiniii I've never even heard of a finalizer.  Cool concept!

**Review by carolyncole — APPROVED** [review]
I ran this branch with puts and main with puts and you can see the http client is created a ton less.  I'm not sure if the finalize is happening though...

https://gist.github.com/carolyncole/54f7219b46ba5f6f0238612f5f0d8287

**Pull request merged**